src/goto-programs/goto_convert_side_effect.cpp

```diff
@@ -112,19 +112,15 @@ void goto_convertt::remove_assignment(
       UNREACHABLE;
     }
 
-    exprt rhs;
-
     const typet &op0_type = to_binary_expr(expr).op0().type();
 
     PRECONDITION(
       op0_type.id() != ID_c_enum_tag && op0_type.id() != ID_c_enum &&
       op0_type.id() != ID_c_bool && op0_type.id() != ID_bool);
 
+    exprt rhs = expr;
```

Comment:
This is not quite the same. The `statement` member will survive this. I'd create a new `binary_exprt`.

Reply:
Hmm, yes, I went too far here. I've now done as you suggested.